Automattic/wp-calypso
# Optimize Calls to Product Endpoint By Allowing Slugs to be Specified to Avoid Loading all Products

Related to # https://github.com/Automattic/martech/issues/2686, https://github.com/Automattic/martech/issues/2722

Calls to the products endpoint load all products by default. This isn't efficient since Calypso does not use all products and on a page refresh the fetch will be done again. It is more efficient to request the required products by slug. 

The `<QueryProductsList />` is the component used to preload/fetch all products. Modifying this component to also include a prop for products slugs will enable us to submit the slugs in the fetch request.

## Proposed Changes

This PR implements 3 improvements: 
1. Modifies `ProductsList.useProducts()` and  `QueryProductsList` component  to accept the required product slugs as a parameter which will be then included on the get request to the endpoint.
2. Modifies  `Site AddOn` and `Difm-landing ` pages to include required product slugs in GET request to products endpoint.
3. Removes `<QueryProductsList />` from Site AddOns page due to the component not being used/needed

## What Areas Benefit 
1. Add-On Fetching 
2. DIFM Landing page

## Testing Instructions

1. Checkout this branch 
4. Open Chrome
5. Open Developer tools and navigate to the Network Tab
6. Filter requests by product
7. Login to Calypso
8. Navigate to Add-Ons
9. In the Network tab you should see individual requests being sent to the products endpoint, however, it should now include the product_slugs query parameter with the slug for the required add-ons.
10. The request time should be a lot faster < 500ms instead of 2-3 seconds
11. Navigate to /start/do-it-for-me/difm-page-picker, complete the flow until payment and confirm it works (all placeholders load)
12. Repeat the above steps using production instance to compare response time difference.

**Point to note:** 
Multiple requests are sent for each add-on. However, these requests are fired concurrently, enabling the maximum response time to be determined by the slowest of all requests

@sirbrillig Adding you as a reviewer since you had [commented](https://github.com/Automattic/wp-calypso/pull/87521) on the first go at this
